cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -0,0 +1,282 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Device object representing Google devices with a grid qubit layout."""
+
+import re
+
+from typing import Any, Set, Tuple, cast
+import cirq
+from cirq_google.api import v2
+
+
+def _validate_device_specification(proto: v2.device_pb2.DeviceSpecification) -> None:
+    """Validates the DeviceSpecification proto.
+
```

Comment:
This docstring doesn't really state much beyond the function name itself.  I would either delete the whole thing (since it is a private function) or add more information.  For instance, you could mention what validations are done and any assumptions made.

Reply:
Ah yeah, since this is private and code comments + error messages already describe what's being validated, I simplified the docstring to just "Raises a ValueError if the DeviceSpecification proto is invalid."